### DISCUSSION TITLE
Access child form fields data from the parent

### DISCUSSION THREAD

**Discussion by tmdoit** [Budibase/budibase#5119] [Help] [answered]
I need to access data from two child forms to build a row and save it to the database.
![Zrzut ekranu 2022-03-28 o 12 01 29](https://user-images.githubusercontent.com/85706044/160378176-f227babd-865a-44bc-b05c-f11d93484258.png)
![Zrzut ekranu 2022-03-28 o 12 01 22](https://user-images.githubusercontent.com/85706044/160378179-36b14384-025c-44fe-b3bd-f833c4cc295e.png)

"Company info" form can be filled automatically with data when using VATNumber search ("Company info" form fields have defined default values which are referenced to the state variables set when requesting data source provider by using search by VAT Number).
I split data in to two forms because in case when someone edit field in "Company info" form field, this edited field will not read data from state variable on resubmitting request (VATNumber search). To support this scenario (update fields data in "Company info" form every time I run VATNumber search), button is executing following actions in order:
- execute query
- clear form
- update state (for each referenced form field)

The problem is, that there are fields which are not related to the request and I don't want to reset their values. ie. contact info like e-mail or phone nr (in attached examples are company address data, "Company address" form)

My thoughts are about accessing child form fields (so that "Save" button can read them and add/update the row). Is it right path to do it? or there is other methodology?

**Comment by melohagan** [comment, marked as answer]
Hi @tmdoit 

For this more complex scenario, it may be easier to use the App State for the entire state of the form. Then instead of having multiple forms and using the *Clear Form* action, use Update State instead to clear the specific fields you want cleared. 

For each form field, the Default value would be set to `{{ State.field_name }}`

**Reply by tmdoit** [reply]
With this approach form fields which are modified/filled with (not default) values will be not replaced by the app state variables (even if modification will leave field empty), that's why I'm clearing the form.
If I open "Create customer" screen (fresh form), fill VAT Number field and hit "Search", "Company name" form field will be filled with data, after when I delete the value or modify it and again hit "Search" button, form field "Company name" will remain untouched, value from app state will not be read again (form field value will not be replaced).

**Reply by melohagan** [reply]
Okay, another possibility could be to split the data collection between two screens. 

So the first screen handles the clearable data. A next button will create the row and then takes you to the next screen via the row id (could be a modal). This _id can be used to update the row just created with the additional data upon save. 
If a cancel button is clicked, then delete the row via the _id and go back to the previous screen.

I know that's probably not ideal, but could be an option.

**Reply by tmdoit** [reply]
I considered this path and others, one idea is to use section element with sidebar and main so when you want to create a customer, steps are following:
1. User goes to "Customers" screen
2. User clicks "Create customer" button
3. System displays form with company data to fill and form to search info by VAT Number
4. User provides VAT Number and hit the search button
5. System fills form data
6. User saves data (add record)
7. System redirects user to customer details page

Customer details page has additional info to fill about the customer, I used section "Sidebar with main" where I use sidebar as an alternative for tabs, example in attached screen. General screen is the equivalent of the form for creating customer,
![Zrzut ekranu 2022-03-29 o 10 58 16](https://user-images.githubusercontent.com/85706044/160574486-7115d695-392d-4d81-9805-7062421f9590.png)

Thanks for your help.
